janus_server/src/aggregator.rs

```diff
@@ -0,0 +1,37 @@
+//! Common functionality for PPM aggregators
+use crate::{
+    hpke::{HpkeRecipient, Label},
+    message::{Role, TaskId},
+};
+use http::{header::CACHE_CONTROL, StatusCode};
+use prio::codec::Encode;
+use std::net::SocketAddr;
+use warp::{reply, trace, Filter};
+
+/// Run a PPM aggregator forever, listening on the provided [`SocketAddr`].
+pub async fn run_aggregator(task_id: TaskId, listen_address: SocketAddr) {
+    let hpke_recipient =
+        HpkeRecipient::generate(task_id, Label::InputShare, Role::Client, Role::Leader);
+
+    let hpke_config_encoded = hpke_recipient.config.get_encoded();
+
+    let hpke_config_endpoint = warp::get()
+        .and(warp::path("hpke_config"))
+        .map(move || {
+            {
+                reply::with_header(
+                    reply::with_status(hpke_config_encoded.clone(), StatusCode::OK),
+                    CACHE_CONTROL,
+                    "max-age=86400",
+                )
+            }
+        })
+        .with(trace::named("hpke_config"));
+
+    let routes = hpke_config_endpoint.with(trace::request());
+
+    println!("aggregator listening on {:?}", listen_address);
```

Comment:
I actually _had_ adopted `tracing` in this PR, but removed that change so that the diff wouldn't be too big, because it does take a few dozen lines of code to install and configure a subscriber. It's not hard to created tracing spans that are annotated with things like a request ID. Anyway, I'll pull in `tracing` and remove the `println`s.

Reply:
We now use `tracing`. Here's sample output from running an aggregator and fetching `hpke_config` (as a screenshot to show off the pretty colors):

![image](https://user-images.githubusercontent.com/19212995/158275088-2ec101e4-2af3-466d-83bf-7769b5eeb24f.png)

Reply:
Thanks, this looks very nice!